tasks/grunt-karma.js

```diff
@@ -137,8 +138,14 @@ module.exports = function (grunt) {
       backgroundProcess.send({ config: data })
       done()
     } else {
-      var server = new Server(data, finished.bind(done))
-      server.start()
+      parseConfig(
```

Comment:
Seems to fail in CI on both Node 10 and Node 12:

> `Running "karma:single" (karma) task`
> `parseConfig(...).then is not a function`

Reply:
Yes - we need to upgrade to Karma 6

Reply:
I'm not sure I follow.

```
    "karma": "^5.0.1",
  },
  "peerDependencies": {
    "karma": "^4.0.0 || ^5.0.0"
```

Does Karma 5 issue a warning that you can't actually fix unless you upgrade to, and require, Karma 6+?

Reply:
I think the warning is only issued by Karma 6, i.e. this patch is only required after we upgrade to Karma 6. I didn't do the upgrade to K6 is this patch as I assumed it would required more work.

I noticed it after https://gerrit.wikimedia.org/r/c/VisualEditor/VisualEditor/+/677879 where package dependency on karma was bumped to 6.

Reply:
OK. I would have expected npm to end up installing karma twice in that patch (once at the top-level, unused; once under `grunt-karma/node_modules`). But for a long time now, npm ignores peerDependencies (other than a soft warning) and requires end-user projects to specify these in their package.json file (if the accept them).

It looks like at least for VisualEditor's case, grunt-karma works fine as-is with karma 6.0, so perhaps it could be added to the `peerDependencies` range. And maybe there's a conditional way to support both for one release here. Then we can merge and release it uncontroversially, without a breaking change that drops karma 4 and 5 (yet).

Reply:
Yeah - we could just check if .parseConfig exists